test/integration/status/test_status.py

```diff
@@ -0,0 +1,341 @@
+
+from utils import (control_socket, extract_status, get_tarantool_version,
+                   pid_file, run_command_and_get_output, run_path, wait_file)
+
+tarantool_major_version, tarantool_minor_version = get_tarantool_version()
+
+
+def start_application(cmd, workdir, app_name, instances):
+    instance_process = subprocess.Popen(
+        cmd,
+        cwd=workdir,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        text=True
+    )
+    start_output = instance_process.stdout.read()
+    for inst in instances:
+        assert f"Starting an instance [{app_name}:{inst}]" in start_output
+
+
+def stop_application(tt_cmd, app_name, workdir):
+    stop_cmd = [tt_cmd, "stop", app_name, "-y"]
+    stop_rc, stop_out = run_command_and_get_output(stop_cmd, cwd=workdir)
+    assert stop_rc == 0
+
+
+def break_config(path):
+    with open(path, "a") as config:
+        config.write("invalid_field: invalid_value\n")
+
+
+def run_command_on_instance(tt_cmd, tmpdir, full_inst_name, cmd):
+    con_cmd = [tt_cmd, "connect", full_inst_name, "-f", "-"]
+    instance_process = subprocess.Popen(
+        con_cmd,
+        cwd=tmpdir,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        text=True
+    )
+    instance_process.stdin.writelines([cmd])
+    instance_process.stdin.close()
+    output = instance_process.stdout.read()
+    return output
+
+
+def wait_instance_status(tt_cmd, tmpdir, full_inst_name, status, port=None, timeout=10):
+    if status == "config":
+        cmd = "return require('config'):info().status"
+        expected_statuses = ["ready", "check_warnings"]
+    elif status == "box":
+        cmd = "return box.info.status"
+        expected_statuses = ["running"]
+    else:
+        raise RuntimeError(f"Not supported status to check: {status}")
+
+    conn = None
+    end_time = time.time() + timeout
+    while True:
+        try:
+            if port:
+                # if socket file doesn't exist use connection by ip:port
+                if not conn:
+                    conn = tarantool.Connection(host="localhost", port=port)
+                res = conn.eval(cmd)[0]
+            else:
+                res = run_command_on_instance(
+                    tt_cmd,
+                    tmpdir,
+                    full_inst_name,
+                    cmd
+                )
+
+            if any(expected_status in res for expected_status in expected_statuses):
+                if conn:
+                    conn.close()
+                return True
+        except tarantool.error.Error:
+            pass
+        if time.time() > end_time:
+            print(f"[{full_inst_name}]: {status} wait timed out after {timeout} seconds.")
+            return False
+        time.sleep(1)
+
+
+@pytest.mark.skipif(tarantool_major_version < 3,
+                    reason="skip cluster instances test for Tarantool < 3")
+@pytest.mark.flaky(reruns=3, reruns_delay=3)
```

Comment:
This is a bad pattern, let's write a working test.

Reply:
Fixed.